airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/FileRecordBuffer.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.record_buffer;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileRecordBuffer implements RecordBufferStorage {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileRecordBuffer.class);
+
+  // The per stream size limit is following recommendations from:
```

Comment:
does this limit also apply to other destinations or is this specific to snowflake?

Reply:
At the moment it applies to all destinations using File buffers. But we could tune the limits differently per destination...

It seems BigQuery has much higher authorized limits:

CSV: Maximum file size - compressed | 4 GB | The size limit for a compressed CSV file is 4 GB.
-- | -- | --
CSV: Maximum file size - uncompressed | 5 TB | The size limit for an uncompressed CSV file is 5 TB.

From https://cloud.google.com/bigquery/quotas#load_jobs

But do we want to use more disk space on the worker to buffer bigger files? 

ATM, I thought it would be fine to use the same "recommended" size limits as suggested by Snowflake as they seem reasonable.

Reply:
> But do we want to use more disk space on the worker to buffer bigger files?
> ATM, I thought it would be fine to use the same "recommended" size limits as suggested by Snowflake as they seem reasonable.

Note that even though the destination warehouse may allow bigger per file size limits. 

This setting would potentially affect how much memory the connector may consume when uploading a large file to the stage area, see https://github.com/airbytehq/airbyte/issues/10260#issuecomment-1073692478